app/resources/cdl/status.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module CDL
+  class Status
+    attr_reader :resource, :user
+    def initialize(resource:, user:)
+      @resource = resource
+      @user = user
+    end
+
+    def as_json(*_args)
+      return no_user if user.blank?
+      {
+        "charged": charged_item.present?,
+        "available": available_status
+      }.merge(expired_hash)
+    end
+
+    def no_user
+      {
+        "charged": false,
+        "available": false
+      }
+    end
+
+    def expired_hash
+      return {} unless charged_item.present?
+      {
+        expires_at: charged_item.expiration_time.to_i
+      }
+    end
+
+    def charged_item
+      charge_manager.charged_items.find do |charged_item|
```

Comment:
suggestion: I wonder if this should be memoized.
It's potentially called three times for every `as_json` call.